source/core/ExecutionContext.cpp

```diff
@@ -140,6 +140,53 @@ VIREO_FUNCTION_SIGNATURE1(FPSync, StringRef)
 #endif
     return _NextInstruction();
 }
+//------------------------------------------------------------
+//
+void SetValueRefNeedsUpdate(TypeRef typeRef, void *dataPtr) {
+    typeRef->SetUpdateNeeded(true);
+}
+
+struct SetValueNeedsUpdateParamBlock : InstructionCore
+{
+    _ParamDef(StaticType, ValueType);
+    _ParamDef(void, ValueData);
+    NEXT_INSTRUCTION_METHOD()
+};
+
+VIREO_FUNCTION_SIGNATURET(SetValueNeedsUpdate, SetValueNeedsUpdateParamBlock)
+{
+    VirtualInstrument* vi = THREAD_EXEC()->_runningQueueElt->OwningVI();
+    if (vi->IsTopLevelVI())
+        SetValueRefNeedsUpdate(_ParamPointer(ValueType), _ParamPointer(ValueData));
+    return _NextInstruction();
+}
+
+Boolean TestValueRefNeedsUpdate(TypeRef typeRef, const void *, Boolean reset) {
+    Boolean needsUpdate = typeRef->IsUpdateNeeded();
+    if (needsUpdate && reset)
+        typeRef->SetUpdateNeeded(false);
+    return needsUpdate;
+}
+
+struct CheckValueNeedsUpdateParamBlock : InstructionCore
+{
+    _ParamDef(StaticType, ValueType);
+    _ParamDef(void, ValueData);
+    _ParamDef(Boolean, NeedsUpdate);
+    NEXT_INSTRUCTION_METHOD()
+};
+VIREO_FUNCTION_SIGNATURET(CheckValueNeedsUpdate, CheckValueNeedsUpdateParamBlock)
```

Comment:
Update the name of the function to CheckValueNeedsUpdateAndReset. This function has the side effect of resetting the NeedsUpdate flag.